src/components/hosts/HostStatus.tsx

```diff
@@ -31,82 +32,55 @@ const getStatusIcon = (status: Host['status']) => {
   if (status === 'insufficient') return <WarningTriangleIcon color={warningColor.value} />;
   if (status === 'installing') return <InProgressIcon />;
   if (status === 'installing-in-progress') return <InProgressIcon />;
+  if (status === 'installing-pending-user-action')
+    return <WarningTriangleIcon color={warningColor.value} />;
   if (status === 'error') return <ExclamationCircleIcon color={dangerColor.value} />;
   if (status === 'installed') return <CheckCircleIcon color={okColor.value} />;
   if (status === 'resetting') return <InProgressIcon />;
   return <UnknownIcon />;
 };
 
-export const getHostInstallationSteps = (role: Host['role'], bootstrap: Host['bootstrap']) => {
-  if (bootstrap) {
-    return [
-      'Starting installation',
-      `Installing as ${role}`,
-      'Bootstrapping installation',
-      'Writing image to disk',
-      'Waiting for control plane',
-      'Rebooting',
-      'Configuring',
-      'Done',
-    ];
-  } else if (role === 'master') {
-    return [
-      'Starting installation',
-      `Installing as ${role}`,
-      'Writing image to disk',
-      'Rebooting',
-      'Configuring',
-      'Joined',
-      'Done',
-    ];
-  } else {
-    return [
-      'Starting installation',
-      `Installing as ${role}`,
-      'Writing image to disk',
-      'Rebooting',
-      'Configuring',
-      'Done',
-    ];
-  }
-};
-
 type HostStatusProps = {
   host: Host;
 };
 
 const HostStatus: React.FC<HostStatusProps> = ({ host }) => {
-  const { status, statusInfo, statusUpdatedAt, role, bootstrap } = host;
+  const { status, statusInfo, statusUpdatedAt } = host;
   const title = HOST_STATUS_LABELS[status] || status;
   const icon = getStatusIcon(status);
+  const hostProgressStages = getHostProgressStages(host);
 
-  const progressInfo = {
-    steps: getHostInstallationSteps(role, bootstrap),
-    currentStep: statusInfo || 'Starting Installation',
-  };
-
-  const currentStepNumber = getHostInstallationStepNumber(
-    progressInfo.steps,
-    progressInfo.currentStep,
-  );
-  const bodyContent = React.useMemo(
-    () => (
-      <div>
-        {['installing', 'installing-in-progress'].includes(status) ? (
-          <>
-            <HostProgress status={status} progressInfo={progressInfo} />
+  const bodyContent = React.useMemo(() => {
```

Comment:
I dont think there's any performance boost from using `useMemo` sice we have `host` in dependency array. And `host` is the only param that can trigger a rerender.

Reply:
I've extracted that into a function